app/Http/Controllers/CategoriesController.php

```diff
@@ -37,19 +38,27 @@ public function show(int $id)
 
     public function update(CategoryRequest $request)
     {
-        $category = Category::query()->findOrFail($request->id);
-        $validatedData = $request->validated();
-        $category->update($validatedData);
+        try {
+            $category = Category::query()->findOrFail($request->id);
+            $validatedData = $request->validated();
+            $category->update($validatedData);
 
-        return redirect()->route('categories.categories')->with('success', 'Categoria editada com successo');
+            return redirect()->route('categories.categories')->with('message', ['success' => 'Categoria editada com successo']);
+        } catch (Throwable $th) {
+            return redirect()->back()->with('message', ['error' => 'Não foi possivel encontrar a categoria']);
+        }
```

Comment:
Se teu objetivo é tratar apenas o not found da categoria, tu pode fazer um catch na exception especifica:
```suggestion
        } catch (ModelNotFoundException) {
            return redirect()->back()->with('message', ['error' => 'Não foi possivel encontrar a categoria']);
        }
```

Mas pq tu precisou tratar esse caso aqui? não da pra deixar que o erro aconteça?